lib/Conversion/TorchToTMTensor/TorchToTMTensor.cpp

```diff
@@ -552,6 +568,270 @@ class ConvertAtenMaxPool2dWithIndicesBackwardOp
+          llvm::to_vector(llvm::map_range(sizes, toOpFoldResult)),
+          llvm::to_vector(llvm::map_range(strides, toOpFoldResult)));
+      offsets[1] =
+          rewriter.createOrFold<arith::AddIOp>(loc, offsets[1], cstOne);
+    }
+    // Cast to static shapes
+    SmallVector<int64_t> staticShape({ShapedType::kDynamic, indexType.getRank()});
+
+    auto staticType = RankedTensorType::get(staticShape, rewriter.getI32Type());
+    indices = rewriter.create<tensor::CastOp>(loc, staticType, indices);
+
+    Value updates = scatterInputsVector[indexType.getRank()];
+
+    Value counts = nullptr;
+    if (reduceType == "mean") {
+        SmallVector<Value> selfShape = getTensorSizes(rewriter, loc, adaptor.getSelf());
+        Attribute initAttr;
+        if (llvm::isa<mlir::FloatType>(srcType.getElementType())) {
+            initAttr = rewriter.getFloatAttr(srcType.getElementType(), 1);
+        } else {
+            initAttr = rewriter.getIntegerAttr(srcType.getElementType(), 1);
+        }
+        Value initElement = rewriter.create<arith::ConstantOp>(loc, initAttr);
+        counts = createInitTensor(rewriter, loc, selfShape,
+                                  selfType.getElementType(), initElement);
+    }
+
+    // If the original values shouldn't be included, normalize the 
+    // input tensor where the scatters take place.
+    if (!includeSelf) {
+      Value normalizationValue;
+      if (reduceType == "sum" || reduceType == "mean") {
+        // Set the values in the input tensor to '0' so they are not included
+        normalizationValue = rewriter.create<arith::ConstantOp>(
+            loc, rewriter.getZeroAttr(srcType.getElementType()));
+      } else if (reduceType == "prod") {
+        // Set the values in the input tensor to '1' (multiplication identity)
+        if (llvm::isa<mlir::FloatType>(srcType.getElementType())) {
+          normalizationValue = rewriter.create<arith::ConstantOp>(
+              loc, rewriter.getFloatAttr(srcType.getElementType(), 1.0));
+        } else {
+          normalizationValue = rewriter.create<arith::ConstantOp>(
+              loc, rewriter.getIntegerAttr(srcType.getElementType(), 1));
+        }
+      } else if (reduceType == "amax") {
+        // Set the values in the input tensor to the smallest element of that type
+        auto minAttr = getNumericLimit(rewriter, srcType.getElementType(), /*getMin=*/true);
+        normalizationValue = rewriter.create<arith::ConstantOp>(loc, minAttr);
+      } else if (reduceType == "amin") {
+        // Set the values in the input tensor to the largest element of that type
+        auto maxAttr = getNumericLimit(rewriter, srcType.getElementType(), /*getMin=*/false);
+        normalizationValue = rewriter.create<arith::ConstantOp>(loc, maxAttr);
+      }
+      Value normalizations = createInitTensor(
+          rewriter, loc, SmallVector<Value>({indexSize}),
+          srcType.getElementType(), /*init_element*/ normalizationValue);
```

Comment:
nit: the format for these comments is `/*init_element=*/normalizationValue`

Reply:
Fixed annotation